src/api/index.test.js

```diff
@@ -12,72 +12,165 @@ limitations under the License.
+    }
+  ]
+};
+
+const serviceAccountFail = {
+  apiVersion: 'v1',
+  kind: 'ServiceAccount',
+  metadata: {
+    creationTimestamp: '2019-08-28T14:50:37Z',
+    name: 'default',
+    namespace: 'tekton-pipelines',
+    resourceVersion: '333247',
+    selfLink: '/api/v1/namespaces/tekton-pipelines/serviceaccounts/default',
+    uid: '3233be5f-c9a3-11e9-a380-025000000001'
+  },
+  secrets: [
+    {
+      name: 'default-token-dcb4b'
+    }
+  ]
+};
+const serviceAccount1 = {
+  apiVersion: 'v1',
+  kind: 'ServiceAccount',
+  metadata: {
+    annotations: {
+      'kubectl.kubernetes.io/last-applied-configuration':
+        '{"apiVersion":"v1","kind":"ServiceAccount","metadata":{"annotations":{},"labels":{"app":"tekton-dashboard"},"name":"tekton-dashboard","namespace":"tekton-pipelines"}}\n'
+    },
+    creationTimestamp: '2019-09-11T09:01:24Z',
+    labels: {
+      app: 'tekton-dashboard'
+    },
+    name: 'tekton-dashboard',
+    namespace: 'tekton-pipelines',
+    resourceVersion: '333387',
+    selfLink:
+      '/api/v1/namespaces/tekton-pipelines/serviceaccounts/tekton-dashboard',
+    uid: 'bb5a9a2c-d472-11e9-a380-025000000001'
+  },
+  secrets: [
+    {
+      name: 'tekton-dashboard-token-6xdwf'
+    },
+    {
+      name: 'secret-name'
+    }
+  ]
+};
+
+const serviceAccount2 = {
+  apiVersion: 'v1',
+  kind: 'ServiceAccount',
+  metadata: {
+    annotations: {
+      'kubectl.kubernetes.io/last-applied-configuration':
+        '{"apiVersion":"v1","kind":"ServiceAccount","metadata":{"annotations":{},"labels":{"app":"tekton-dashboard"},"name":"tekton-dashboard","namespace":"tekton-pipelines"}}\n'
+    },
+    creationTimestamp: '2019-09-11T09:01:24Z',
+    labels: {
+      app: 'tekton-dashboard'
+    },
+    name: 'Groot',
+    namespace: 'tekton-pipelines',
+    resourceVersion: '333387',
+    selfLink:
+      '/api/v1/namespaces/tekton-pipelines/serviceaccounts/tekton-dashboard',
+    uid: 'bb5a9a2c-d472-11e9-a380-025000000001'
+  },
+  secrets: [
+    {
+      name: 'tekton-dashboard-token-6xdwf'
+    }
+  ]
+};
+
+const serviceAccountList = { items: serviceAccountFail };
+const serviceAccountListMultipleWithNoSecretName = {
+  items: serviceAccountFail,
+  serviceAccountPass
+};
+const serviceAccountListMultiple = {
+  items: serviceAccountFail,
+  serviceAccountPass,
+  serviceAccountPassV2
+};
+
 describe('getAPIRoot', () => {
   it('handles base URL with trailing slash', () => {
     window.history.pushState({}, 'Title', '/path/#hash');
-    expect(getAPIRoot()).toContain('/path');
+    expect(index.getAPIRoot()).toContain('/path');
   });
 
   it('handles base URL without trailing slash', () => {
     window.history.pushState({}, 'Title', '/path#hash');
-    expect(getAPIRoot()).toContain('/path');
+    expect(index.getAPIRoot()).toContain('/path');
   });
 });
 
 describe('getAPI', () => {
   it('returns a URI containing the given type', () => {
-    const uri = getAPI('pipelines');
+    const uri = index.getAPI('pipelines');
```

Comment:
why is `index` prefixing now required?

Reply:
The tests in jest require something to be in from the function to be mocked like 'index'
Which meant having to import * from index as index instead of individual functions so we had to put index. in front of them
It's how its done in the rest of the code and saw no way around doing it that way